salt/cloud/__init__.py

```diff
@@ -185,6 +185,10 @@ def __init__(self, path=None, opts=None, config_dir=None, pillars=None):
         else:
             self.opts = salt.config.cloud_config(path)
 
+        # Check the cache-dir exists. If not, create it.
+        if not os.path.exists(self.opts['cachedir']):
```

Comment:
I'm very curious about the underlying cause of this issue. Under what cases have you seen this occur?

Reply:
Totally fair. Thanks for the detailed explanation.